src/Microsoft.Dnx.Tooling/Utils/PackageFeedCache.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Concurrent;
+using Microsoft.Dnx.Tooling.Restore.NuGet;
+using NuGet;
+
+namespace Microsoft.Dnx.Tooling
+{
+    public class PackageFeedCache
+    {
+        private ConcurrentDictionary<string, HttpSource> _httpSources = new ConcurrentDictionary<string, HttpSource>();
+
+        public IPackageFeed GetPackageFeed(PackageSource source, bool noCache, bool ignoreFailedSources,
+            Reports reports)
+        {
+            // Check if the feed is a URL
+            if (source.IsLocalFileSystem())
+            {
+                return PackageFolderFactory.CreatePackageFolderFromPath(source.Source, ignoreFailedSources, reports);
```

Comment:
Local repos cache results of lookups -- https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Tooling/NuGet/Core/Repository/LocalPackageRepository.cs#L19-L20

Would be useful to reuse these instances too